work_queue/src/work_queue.c

```diff
@@ -4784,6 +4789,30 @@ void work_queue_task_specify_algorithm(struct work_queue_task *t, work_queue_sch
 	t->worker_selection_algorithm = algorithm;
 }
 
+void work_queue_task_specify_conda_env(struct work_queue_task *t, const char *local_tarball) {
+	assert(t);
+	assert(local_tarball);
+
+	if(t->conda_environment) {
+		fatal("A conda environment for the task has been already specified.");
+	}
+
+	work_queue_task_specify_file(t, local_tarball, path_basename(local_tarball), WORK_QUEUE_INPUT, WORK_QUEUE_CACHE);
```

Comment:
Would it make sense to require the user to explicitly call `specify_file` and then `conda_env` separately.
This will permit the tarball to arrive in different ways, using the current (and future) capabilities of `specify_file`.
This would also resolve the asymmetry in the worker below.

Reply:
Let's try that.  Specifying the file twice did not seem natural to me at first, but the more I think about it, it matches better with the rest of the API.